src/core/link/Link.tsx

```diff
@@ -54,8 +56,10 @@ export function Link({
       href={href}
       openInNewTab={openInNewTab ?? isExternal}
       external={isExternal}
-      showExternalIcon={showExternalIcon ?? isExternal}
-      className={classNames(styles.link, className)}
+      showExternalIcon={showExternalIcon ?? (isExternal && !hasImageInLink)}
```

Comment:
How about moving this also into isExternal?
const isExternal = getIsHrefExternal(href) && !isEmail && !isPhone;
->
const isExternal = getIsHrefExternal(href) && !isEmail && !isPhone && !hasImageInLink;